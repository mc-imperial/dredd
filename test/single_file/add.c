int main() {
  int x = 1;
  int y = 2;
  return x + y + x;
}
