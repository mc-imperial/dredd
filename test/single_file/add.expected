#include <cstdlib>
#include <functional>

static int __dredd_enabled_mutation() {
  static bool initialized = false;
  static int value;
  if (!initialized) {
    const char* __dredd_environment_variable = std::getenv("DREDD_ENABLED_MUTATION");
    if (__dredd_environment_variable == nullptr) {
      value = -1;
    } else {
      value = atoi(__dredd_environment_variable);
    }
    initialized = true;
  }
  return value;
}

static int __dredd_replace_binary_operator_Add_int_int(std::function<int()> arg1, std::function<int()> arg2, int mutation_id) {
  switch (__dredd_enabled_mutation() - mutation_id) {
    case 0: return arg1() / arg2();
    case 1: return arg1() * arg2();
    case 2: return arg1() % arg2();
    case 3: return arg1() - arg2();
    case 4: return arg1();
    case 5: return arg2();
    default: return arg1() + arg2();
  }
}

int main() {
  int x = 1;
  int y = 2;
  if (__dredd_enabled_mutation() != 12) { return __dredd_replace_binary_operator_Add_int_int([&]() -> int { return static_cast<int>(__dredd_replace_binary_operator_Add_int_int([&]() -> int { return static_cast<int>(x); }, [&]() -> int { return static_cast<int>(y); }, 0)); }, [&]() -> int { return static_cast<int>(x); }, 6); }
}
